web/server/api/endpoints/plan.py

```diff
@@ -2,6 +2,7 @@
 
 import asyncio
 import typing as t
+from time import sleep
```

Comment:
Why do we need all the sleeps?

Reply:
sleeps  just to slow down state change a bit so UI can transition smoother. I can delay it on UI but it will involve touching many places and adding bunch of `setTimeout`s and `useState`

Reply:
that should definitely not be done in the api